src/app/auth/auth.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable, BehaviorSubject } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { LoginResponse } from '../models/login-response.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  private apiUrl = 'http://localhost:8080/api/v1/auth';
+  private isAuthenticatedSubject = new BehaviorSubject<boolean>(false);
+
+  constructor(private http: HttpClient) { }
+
+  login(username: string, password: string): Observable<any> {
```

Comment:
或許你可以考慮把 username: string, password: string 抽出來變成一個 model。然後 Response 我看你好像有 LoginResponse?

```suggestion
  login(LoginRequest): Observable<LoginResponse> {
```